default.nix

```diff
@@ -43,6 +43,42 @@ in
                  url = "http://hackage.haskell.org/package/${pkgver}/${pkgver}.tar.gz";
                  inherit sha256;
                }) {};
+
+          # Adds a nix file as an input to the haskell derivation it
+          # produces. This is useful for callHackage / callCabal2nix to
+          # prevent the generated default.nix from being garbage collected
+          # (requiring it to be frequently rebuilt), which can be an
+          # annoyance.
+          callPackageKeepDeriver = src: args:
+            overrideCabal (self.callPackage src args) (orig: {
+              preConfigure = ''
+                # Generated from ${src}
+                ${orig.preConfigure or ""}
+              '';
+              passthru = orig.passthru or {} // {
+                # When using callCabal2nix or callHackage, it is often useful
+                # to debug a failure by inspecting the Nix expression
+                # generated by cabal2nix. This can be accessed via this
+                # cabal2nixDeriver field.
+                cabal2nixDeriver = src;
+              };
+            });
+
+          callCabal2nixWithOptions = name: src: extraCabal2nixOptions: args:
```

Comment:
This function exists in `nixpkgs` since August of last year, but it seems to be missing from our particular pin. Without this, we cannot pin to GIthub-based dependencies that exist as an "uber repo" (i.e. no `.cabal` at the top level).